status/views.py

```diff
@@ -1,97 +1,109 @@
-                reasons.append(f"High CPU load: {sysinfo['Loads'][0]}")
-            if sysinfo['FreeRam'] / sysinfo['TotalRam'] < 0.2:
-                reasons.append(f"High RAM usage: {sysinfo['FreeRam'] / 1024}MB free out of {sysinfo['TotalRam']}")
-            if len(reasons) != 0:
-                data[2]['status'] = 'partial'
-                data[2]['details'] = '\n'.join(reasons)
+
+class StatusView(APIView):
+    permission_classes = (IsAdminUser,)
+
+    def get(self, request):
+        connection_info = []
+
+        redis_status, redis_details = self.test_redis_connection()
+        connection_info.append({"name": "Redis", "status": redis_status, "details": redis_details})
+
+        postgres_status, postgres_details = self.test_postgresql_connection()
+        connection_info.append({"name": "PostgreSQL", "status": postgres_status, "details": postgres_details})
+
+        andromeda_status, andromeda_details = self.get_andromeda_status()
+        connection_info.append({"name": "Andomeda", "status": andromeda_status, "details": andromeda_details})
+
+        if settings.SEND_MAIL:
+            mailusv_status, mailusv_details = self.test_mailusv_connection()
+            connection_info.append({"name": "Mail Daemon", "status": mailusv_status, "details": mailusv_details})
+
+        return FormattedResponse(connection_info)
+
+    def test_redis_connection(self):
+        status, details = 'unknown', ''
+
+        try:
+            rs_details = settings.CONFIG['REDIS']
+            rs = redis.Redis(host=rs_details['HOST'], port=rs_details['PORT'])
+
+            if rs.ping() == True:
+                status = 'online'
             else:
-                data[2]['status'] = 'online'
-        else:
-            data[2]['status'] = 'unknown'
-            data[2]['details'] = 'Error connecting to /status'
-    except Exception as e:
-        data[2]['status'] = 'offline'
-        data[2]['details'] = str(e)
-
-    if settings.SEND_MAIL:
-        data.append(
-        {
-            "name": "Mail Daemon",
-            "status": "unknown",
-            "details": ''
-        })
+                status = 'offline'
+                details = "PING failed"
+        except redis.exceptions.RedisError as e:
+            status = 'offline'
+            details = "PING failed:\n" + str(e)
+
+        return status, details
+
+    def test_postgresql_connection(self):
+        status, details = 'unknown', ''
+
         try:
-            response = requests.get(settings.MAIL_SOCK_URL)
-            if response.status_code != 200:
-                data[3]['status'] = 'unknown'
-                data[3]['details'] = 'Unknown error whilst connecting to /'
+            with connection.cursor() as cursor:
+                cursor.execute("SELECT 1");
+            status = 'online'
+        except psycopg2.OperationalError as e:
+            status = 'offline'
+            details = 'Could not execute query `SELECT 1\':\n' + str(e)
```

Comment:
It could be worth separating the exception from the main part of the details, if someones looking at the status page, the fact postgres is dead is much more significant than the actual exception and off the top of my head, str(e) isn't gonna give you a whole lot of useful info anyway.

Reply:
In the status page it comes up with an offline icon and then the details are put in a code block below and when I did `str(e)` It was more the fact that you could tell if it was postgres still starting up (which happened to me) or if postgres was giving you an error directly etc.